integrationtests/storage/integrationtest_storage.py

```diff
@@ -122,15 +125,59 @@ def check_data_per_worker() -> None:
 
     storage = StorageStub(storage_channel)
 
-    for worker_id in range(6):
-        request = GetDataPerWorkerRequest(dataset_id="test_dataset", worker_id=worker_id, total_workers=6)
+    # 30 images in total; the first two workers should get 5 images each, the rest should get 4 images
+    for worker_id in range(7):
+        request = GetDataPerWorkerRequest(dataset_id="test_dataset", worker_id=worker_id, total_workers=7)
+        responses: list[GetDataPerWorkerResponse] = list(storage.GetDataPerWorker(request))
+
+        assert len(responses) == 1, f"Received batched response or no response, shouldn't happen: {responses}"
+
+        response_keys_size = len(responses[0].keys)
+
+        assert response_keys_size == 5 if worker_id <= 1 else response_keys_size == 4
+
+    split_ts1 = IMAGE_UPDATED_TIME_STAMPS[9] + 1
+    split_ts2 = IMAGE_UPDATED_TIME_STAMPS[19] + 1
+
```

Comment:
The rest tests are against `GetDataPerWorkerRequest` with `start_timestamp` or `end_timestamp` set.